### PR TITLE
Uninstalled docisfy, installed mkdocs

### PR DESCRIPTION
**Pull request by dev-matan-tsuberi**
This PR removes [docisfy](https://docsify.js.org) in favor of [mkdocs](http://www.mkdocs.org/) + [mkdocs-material](https://github.com/squidfunk/mkdocs-material) documentation website.

[**Live example + Instructions on how to run locally/deploy**](https://dev-matan-tsuberi.github.io/daostack/)

`mkdocs` is a python packcage that generates static html from `.md`.
`mkdocs-material` is a material design theme for `mkdocs`.

Changes:
- removed all traces of `docsify` (`docs/index.html`, `docs/.nojekyll`).
- `mkdocs.yml` - is the config file for `mkdocs`.
- Added commands as described [here](https://dev-matan-tsuberi.github.io/daostack/#contributing-to-arc-docs) to `package.json`
- renamed all `docs/**/README.md` to `index.md` so they appear as homepages for their dir.
- updated some links.
- expanded `guides.md` to it's own directory with a couple of stub articles.
- removed `docs/ref` - we don't need it on `master`, it's automatically generated and pushed to `gh-pages` branch instead, using `yarn run docs:deploy`. (and it's called `reference` now).
- moved prehistoric docs to `docs_old` where they will stay until revised as per 
 https://github.com/daostack/Arc/issues/272
- moved `docs/headers` to `docs_headers` to ease the build process a bit.

**Further prep:**
- Please enable github pages from `gh-pages` branch instead of `master/docs` in `settings`.

Pros:
- Search feature on all the docs.
- Built in table of contents, menu, social links (slack,telegram,gh).
- Much cleaner & clearer website.
- Generated docs no longer live on `master` = less clutter.
- More customization.

**Note: If this PR merges, i'll close https://github.com/daostack/Arc/pull/280*

### CODE REVIEW

**Comment by orenyodfat** [comment]
It has an issue on Safari 

screen shot 2018-01-30 at 0 03 24

**Comment by dev-matan-tsuberi** [comment]
@orenyodfat 
Safari was discontinued on windows since 2016 (according to [this](https://apple.stackexchange.com/questions/68836/where-can-i-download-safari-for-windows)), using the latest version available for windows I saw a different bug, but not the one you saw. So I can't really reproduce it, you can post an issue [here](https://github.com/squidfunk/mkdocs-material).

**Comment by orenyodfat** [comment]
@dev-matan-tsuberi  as safari is used by around 3.5% of the browser users I guess we can live with that (though it raise a question about the quality of this product). did you tried that on firefox ?

File: docs/guides/building_a_simple_charity.md [hunk 1]
```diff
@@ -0,0 +1 @@
+TODO ...
```
Comment (orenyodfat) [inline docs/guides/building_a_simple_charity.md]: What is the charity example ?

**Comment by dev-matan-tsuberi** [comment]
@orenyodfat works fine on the latest firefox.

**Pull request merged**